types/pair/README.md

```diff
@@ -0,0 +1,15 @@
+# `std::pair`
+
+## Fields
+
+ * `Int32_String`: `std::pair<std::int32_t, std::string>`
+ * `Variant_Vector`: `std::pair<std::variant<std::int32_t, std::string>, std::vector<std::int32_t>>`
```

Comment:
Would also testing `std::optional` add anything?

Reply:
Nullable fields have the structural role of a collection, so from the binary format point of view this case is covered with `std::vector`.

Reply:
Not saying I disagree, but this is a bit contradictory to what you wrote in https://github.com/root-project/rntuple-validation/pull/39#issuecomment-2697930196...